cfgov/prepaid_agreements/views.py

```diff
@@ -0,0 +1,121 @@
+from django.contrib.postgres.search import SearchVector
+from django.core.paginator import InvalidPage, Paginator
+from django.db.models import Q
+from django.shortcuts import render
+
+from prepaid_agreements.models import PrepaidProduct
+
+
+def validate_page_number(request, paginator):
+    """
+    A utility for parsing a pagination request,
+    catching invalid page numbers and always returning
+    a valid page number, defaulting to 1.
+    """
+    raw_page = request.GET.get('page', 1)
+    try:
+        page_number = int(raw_page)
+    except ValueError:
+        page_number = 1
+    try:
+        paginator.page(page_number)
+    except InvalidPage:
+        page_number = 1
+    return page_number
+
+
+def search_products(search_term, search_field, products):
+    search_fields = [
+        'issuer_name',
+        'other_relevant_parties',
+        'name',
+        'program_manager',
+        'prepaid_type',
+    ]
+    if search_field and search_field[0] in search_fields:
+        search_fields = [search_field[0]]
+    products = products.annotate(
+        search=SearchVector(
```

Comment:
Hmm, yeah I just ran into this while writing tests for the search.   I'm a little conflicted on what to do - this feels like a test configuration issue, and usage of `SearchVector` seems like the way to go here, but I could also rewrite these queries to not be dependent on Postgres.  I'd prefer the former but that would involve a change in how we all run tests.  Curious what your thoughts are and if other BEWDs have a preference here -- @willbarton @higs4281 @schbetsy ?

Reply:
Based on feedback in a BEWD meeting today, I'll be leaving this code as-is and the tests will now need to be run against Postgres locally by passing in a TEST_DATABASE_URL like so: 

TEST_DATABASE_URL=postgres://django@localhost/cfgov tox -e fast

I will update the documentation.

Reply:
Is your intention to recommend that developers always use a local Postgres when running tests, or to suggest using Postgres only when you want to run these new specific tests?

If the former, it would be better to update [the existing `settings.test` code](https://github.com/cfpb/cfgov-refresh/blob/fb96d5ef92df6068a4659911fc350da6734aaa1e/cfgov/cfgov/settings/test.py#L4-L17) which currently defaults to a Sqlite DB, although this introduces some complexity see below).

Thinking about this some more, an even better option might be to leave the default test DB as Sqlite but mark these Postgres-specific tests with `@unittest.skipUnless` (like [this example](https://github.com/django/django/blob/193c109327c5216cab1d4eed6bfdff24629b09a3/tests/postgres_tests/__init__.py#L9) in the Django code) so that they only execute if being run against Postgres. This would have the advantage of making it easier to run tests without requiring users to set up Postgres, but still let us test against Postgres on Travis. The docs also [already suggest](https://cfpb.github.io/cfgov-refresh/python-unit-tests/#running-tests) that you can override `TEST_DATABASE_URL` if you want to test locally against Postgres.

Back to your specific proposal for `TEST_DATABASE_URL`, one problem is that we can't hardcode `localhost` as the host, because that won't work for people running tests in the Docker setup, where the Postgres host is named `postgres`. So this value would have to be different for the two different methods of running, which probably means adding `TEST_DATABASE_URL` to the [.env_SAMPLE](https://github.com/cfpb/cfgov-refresh/blob/master/.env_SAMPLE) as we have `DATABASE_URL`.

Because of the additional complexity this would add, I would prefer enabling the tests so that they run only when run against Postgres.

Reply:
@chosak I didn't realize I could skip them like that -- that's perfect, thanks for the suggestion.  I've implemented this in https://github.com/cfpb/cfgov-refresh/commit/34ae7b516896972c1e6ddd0d6311d9eded8a9018